src/lib/templateengine.js

```diff
@@ -274,33 +276,20 @@ function TemplateEngine( undefined ) {
   
   this.enableAddressQueue = $.getUrlVar('enableQueue') ? true : false;
   
-  this.backend = 'cgi-bin'; // default path to backend
+  this.backend = 'default';
+  this.backendUrl = null;
```

Comment:
As I said, it's not wrong to use `null`. See e.g. http://stackoverflow.com/questions/5076944/what-is-the-difference-between-null-and-undefined-in-javascript for a deeper discussion. The accepted answer has IMHO a good conclusion:
- `undefined` means a variable has been declared but has not yet been assigned a value
- `null` is an assignment value. It can be assigned to a variable as a representation of no value

I'm fine with `null` when that's the logical value here. When you want to say "default" for being not set, then in my opinion `undefined` would be the logical choice.